presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -474,7 +476,11 @@ private void validateSchema(Optional<String> partitionName, Properties schema)
                 partitionValues.add(HIVE_DEFAULT_DYNAMIC_PARTITION);
             }
             else {
-                partitionValues.add(value.toString());
+                String valueString = value.toString();
+                if (!CharMatcher.inRange((char) 0x20, (char) 0x7E).matchesAllOf(valueString)) {
+                    throw new PrestoException(HIVE_INVALID_PARTITION_VALUE, "\"" + valueString + "\" contains characters not in the range of 0x20 - 0x7E");
```

Comment:
Error message need to mention partition key and Hive. Users don't generally see error code. And it is important that the error message provides user with enough information.